tests/unit_tests/wipeable_string.cpp

```diff
@@ -0,0 +1,181 @@
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <string.h>
+#include "gtest/gtest.h"
+
+#include "wipeable_string.h"
+
+TEST(wipeable_string, ctor)
+{
+  epee::wipeable_string s0;
+  ASSERT_EQ(s0.size(), 0);
+
+  epee::wipeable_string s1(std::string("foo"));
+  ASSERT_EQ(s1.size(), 3);
+  ASSERT_TRUE(!memcmp(s1.data(), "foo", s1.size()));
+
+  epee::wipeable_string s2(std::string("bar"));
+  ASSERT_EQ(s2.size(), 3);
+  ASSERT_TRUE(!memcmp(s2.data(), "bar", s2.size()));
+
+  epee::wipeable_string s3(std::string("quux"));
+  ASSERT_EQ(s3.size(), 4);
+  ASSERT_TRUE(!memcmp(s3.data(), "quux", s3.size()));
+}
+
+TEST(wipeable_string, wipe)
+{
+  epee::wipeable_string s0(std::string("foo"));
+  ASSERT_EQ(s0.size(), 3);
+  s0.wipe();
+  ASSERT_EQ(s0.size(), 3);
+  ASSERT_TRUE(!memcmp(s0.data(), "\0\0\0", 3));
+}
+
+TEST(wipeable_string, clear)
+{
+  epee::wipeable_string s0(std::string("foo"));
+  ASSERT_EQ(s0.size(), 3);
+  s0.clear();
+  ASSERT_EQ(s0.size(), 0);
+}
+
+TEST(wipeable_string, push_back)
+{
+  epee::wipeable_string s0(std::string("fo"));
+  ASSERT_EQ(s0.size(), 2);
+  s0.push_back('o');
+  ASSERT_EQ(s0.size(), 3);
+  ASSERT_TRUE(!memcmp(s0.data(), "foo", s0.size()));
+}
+
+TEST(wipeable_string, append_char)
+{
+  epee::wipeable_string s0(std::string("fo"));
+  ASSERT_EQ(s0.size(), 2);
+  s0 += 'o';
+  ASSERT_EQ(s0.size(), 3);
+  ASSERT_TRUE(!memcmp(s0.data(), "foo", s0.size()));
+}
+
+TEST(wipeable_string, append_string)
+{
+  epee::wipeable_string s0(std::string("foo"));
+  ASSERT_EQ(s0.size(), 3);
+  s0 += "bar";
+  ASSERT_EQ(s0.size(), 6);
+  ASSERT_TRUE(!memcmp(s0.data(), "foobar", s0.size()));
+}
+
+TEST(wipeable_string, empty)
+{
+  epee::wipeable_string s0;
+  ASSERT_TRUE(s0.empty());
+  s0.push_back(' ');
+  ASSERT_FALSE(s0.empty());
+  s0.pop_back();
+  ASSERT_TRUE(s0.empty());
+}
+
+TEST(wipeable_string, pop_back)
+{
+  epee::wipeable_string s = "test";
+  ASSERT_EQ(s.size(), 4);
+  s.pop_back();
```

Comment:
Maybe also check for the returned value?

Reply:
OK